src/lib/parser.rs

```diff
@@ -238,25 +247,30 @@ impl Parser {
       return Ok(());
     }
     self.index -= 1;
-    match self.try_match(&[(Keywords::Fn, " \t\n"), (Keywords::Const, " \t\n")]) {
-      Some(Keywords::Const) => {
-        let parsed_variable = parse_var(self, Some(VarType::Const))?;
-        self.global_vars.push(parsed_variable);
-        Ok(())
-      }
-      Some(Keywords::Fn) => {
-        let parsed_function = ParseFunction::start(self)?;
-        self.functions.push(parsed_function);
-        Ok(())
-      }
-      _ => {
-        if let Some(c) = self.next_char() {
-          self.unexpected_char(c)
-        } else {
-          self.unexpected_eof()
+    while self.index < self.contents.len() {
+      match self.try_match(&[(Keywords::Fn, " \t\n"), (Keywords::Const, " \t\n")]) {
+        Some(Keywords::Const) => {
+          let parsed_variable = parse_var(self, Some(VarType::Const))?;
+          self.global_vars.push(parsed_variable);
+        }
+        Some(Keywords::Fn) => {
+          let parsed_function = ParseFunction::start(self)?;
+          self.functions.push(parsed_function);
+        }
+        _ => {
+          // could be newline/tab/whitespace
+          if let Some(c) = self.next_char() {
+            if c == '\n' || c == '\t' || c == ' ' {
+              continue;
+            }
+            return self.unexpected_char(c)
+          } else {
+            return self.unexpected_eof()
+          }
         }
       }
     }
+    Ok(())
```

Comment:
I just tested it using this code and the test results where the same:  
```suggestion
    while let Some(_) = self.next_while(" \n\t") {
      self.index -= 1;
      match self.try_match(&[(Keywords::Fn, " \t\n"), (Keywords::Const, " \t\n")]) {
        Some(Keywords::Const) => {
          let parsed_variable = parse_var(self, Some(VarType::Const))?;
          self.global_vars.push(parsed_variable);
        }
        Some(Keywords::Fn) => {
          let parsed_function = ParseFunction::start(self)?;
          self.functions.push(parsed_function);
        }
        _ => {
          // could be newline/tab/whitespace
          if let Some(c) = self.next_char() {
            return self.unexpected_char(c);
          } else {
            return self.unexpected_eof();
          }
        }
      }
    }
```
Note that i've removed the `if let None = self.next_while(" \n\t") {` check above here